typing/typecore.ml

```diff
@@ -3775,15 +3794,24 @@ and type_expect_
           let mode = Value_mode.newvar () in
           mode, mode_nontail mode
       in
-      let rec lower_args seen ty_fun =
+      (* does the function return a tvar which is too generic? *)
+      let rec ret_tvar seen ty_fun =
         let ty = expand_head env ty_fun in
-        if TypeSet.mem ty seen then () else
+        if TypeSet.mem ty seen then false else
           match get_desc ty with
             Tarrow (_l, ty_arg, ty_fun, _com) ->
               (try unify_var env (newvar()) ty_arg
                with Unify _ -> assert false);
-              lower_args (TypeSet.add ty seen) ty_fun
-          | _ -> ()
+              ret_tvar (TypeSet.add ty seen) ty_fun
+          | Tvar _ -> 
+              let v = newvar () in 
+              let rt = get_level ty > get_level v in 
```

Comment:
Here `ret_tvar` is computed as `get_level ty > get_level v`, but in the old code it was `get_level ty_fun >= get_level ty_arg`. Is the change from `>` to `>=` intentional? (I don't really get what's going on with this change)

Reply:
yes it is intentional, because `get_level v` is one less now, because the original code has L3825

Reply:
Thanks! Missed that.